core/cache/resident.go

```diff
@@ -50,17 +49,30 @@ type residentManager struct {
 	residentCount *counter
 	resourceCount *counter
 
+	// residents are all the residents of the cache indexed by ID.
+	// Access to this map should be Goroutine-safe.
```

Comment:
Access to this map needs to be done using 'mu' to ensure it is goroutine safe.

The way it is written sounds like just accessing the map will be fine, but while it might be ok outside this type because we make sure to use the mutex, it *is* something someone working on this type needs to watch out for.